vim-patch:8.1.{140,1269},8.2.0161

janlazo opened the pull request.

janlazo committed:
vim-patch:8.2.0161: not recognizing .gv file as dot filetype

Problem:    Not recognizing .gv file as dot filetype.
Solution:   Add *.gv to dot pattern. (closes vim/vim#5544)
https://github.com/vim/vim/commit/f8ddb25789a6af530e69f499907979dfbff1c1ea

justinmk commented:
One test failure, but maybe the new behavior is correct?

```
[  FAILED  ] test/functional/terminal/mouse_spec.lua @ 186: :terminal mouse when the terminal has focus with a split window and other buffer handles terminal size when switching buffers
test/functional/terminal/mouse_spec.lua:199: Row 1 did not match.
Expected:
  |*rows: 5, cols: 25        │rows: 5, cols: 25       |
  |*rows: 5, cols: 24        │rows: 5, cols: 24       |
  |*mouse enabled            │mouse enabled           |
  |*rows: 5, cols: 25        │rows: 5, cols: 25       |
  |*{2:^ }                        │{2: }                       |
  |==========                ==========              |
  |:bn                                               |
Actual:
  |*^line21                   │rows: 5, cols: 25       |
  |*line22                   │rows: 5, cols: 24       |
  |*line23                   │mouse enabled           |
  |*line24                   │rows: 5, cols: 25       |
  |*line25                   │{2: }                       |
  |==========                ==========              |
  |:bn                                               |
```

janlazo commented:
8.1.0425 was suppose to fix 8.1.0342 which fixes 8.0.1732. I expected 0 test failures 

I'll leave 8.1.0425 WIP until the other (terminal) patches are merged. Need to resolve https://github.com/neovim/neovim/pull/10396 before continuing on the other terminal patches.

8.2.0161 is safe to cherry-pick.

janlazo committed:
vim-patch:8.1.0140: recording into a register has focus events

Problem:    Recording into a register has focus events. (Michael Naumann)
Solution:   Don't record K_FOCUSGAINED and K_FOCUSLOST. (closes vim/vim#3143)
https://github.com/vim/vim/commit/972bfddc6b3f52ae0865ad8c0bf6089bc8a9883a

janlazo commented on src/nvim/keymap.h:
```diff
@@ -240,8 +240,8 @@ enum key_extra {
-  , KE_FOCUSGAINED = 98       // focus gained
-  , KE_FOCUSLOST = 99         // focus lost
+  // , KE_FOCUSGAINED = 98    // focus gained
+  // , KE_FOCUSLOST = 99      // focus lost
```
These are used in the original vim patch for `gotchars()` but Neovim removed `K_FOCUS` pseudo-keys.

janlazo commented on src/nvim/getchar.c:
```diff
@@ -1096,26 +1096,40 @@ void del_typebuf(int len, int offset)
+    }
+    if (buflen == 2) {
+      continue;
+    }
```
Omitted the `buflen == 3` condition because Neovim doesn't support `K_FOCUS` pseudo keys.

janlazo committed:
vim-patch:8.1.1269: MS-Windows GUI: multibyte chars with a 0x80 byte do not work

Problem:    MS-Windows GUI: multibyte chars with a 0x80 byte do not work when
            compiled with VIMDLL.
Solution:   Adjust the condition for fixing the input buffer. (Ken Takata,
            closes vim/vim#4330)
https://github.com/vim/vim/commit/ed5ab2a95972b5ef588bdafab9f197e1dcf0c1df

justinmk reviewed (approved):
waiting on CI
